hw/xfree86/common/xf86DGA.c

```diff
@@ -1783,17 +1754,18 @@ ProcXF86DGAGetVideoLL(ClientPtr client)
     /* get the parameters for the mode that best matches */
     DGAGetModeInfo(stuff->screen, &mode, num);
 
+    xXF86DGAGetVideoLLReply rep = {
+        .offset = mode.offset,
+        .width = mode.bytesPerScanline / (mode.bitsPerPixel >> 3),
+        .ram_size = rep.bank_size >> 10
+    };
+
     if (!DGAOpenFramebuffer(stuff->screen, &name,
                             (unsigned char **) (&rep.offset),
                             (int *) (&rep.bank_size), &offset, &flags))
         return BadAlloc;
 
-    rep.offset += mode.offset;
-    rep.width = mode.bytesPerScanline / (mode.bitsPerPixel >> 3);
-    rep.ram_size = rep.bank_size >> 10;
```

Comment:
I’m not sure how `DGAOpenFramebuffer` works, but if it writes to `rep.offset` and `rep.bank_size` we may need to keep these removed lines after the `DGAOpenFramebuffer` call as it was originally.